client/components/email-verification/email-verification-gate.jsx

```diff
@@ -76,3 +49,14 @@ export default class EmailVerificationGate extends React.Component {
 		return <div>{ this.props.children }</div>;
 	}
 }
+
+export default connect(
+	( state ) => {
+		const user = getCurrentUser( state );
+		const site = getSelectedSite( state );
+		return {
+			userEmail: user && user.email,
+			needsVerification: userUtils.needsVerificationForSite( site )
```

Comment:
It feels a little odd to rely on `userUtils` here in a pure component. Since it'll attempt to fetch the current user object before checking whether verification is needed, do you think there can be a case where this could go wrong?